### ISSUE TITLE
Coefficients changing depending on the Taylor order

### ISSUE THREAD

**Issue by matthieubulte** [JuliaDiff/TaylorSeries.jl#275]
Hi, I just came across the following issue,

```julia
# If I take a series of 3rd order, everything works well
julia> t = Taylor1(Float64, 3)
 1.0 t + 𝒪(t⁴)

julia> log((exp(t/2) - exp(-t/2))/t)
 0.041666666666666664 t² + 𝒪(t⁴)

# But for a series of second order, the coefficient of the quadratic term is suddenly 0
julia> t = Taylor1(Float64, 2)
 1.0 t + 𝒪(t³)

julia> log((exp(t/2) - exp(-t/2))/t)
 0.0 + 𝒪(t³)
```

I do understand that this expression is tricky in `0`, but since the third order expansion handled it well, I was surprised to see that the second order expansion failed.

**Comment by lbenet** [comment]
Thanks for reporting! 

This issue is somewhat related to #230 (see also #226). The problem is that factorizing `t` *lowers*  the order of the resulting Taylor expansion, which is not what currently is being done; this is addressed in #248. 

If you do the same using the proposed changes in #248, you get consistent results.
```julia
# third order expansion
julia> t = Taylor1(Float64, 3)
 1.0 t + 𝒪(t⁴)

julia> exp(t/2) - exp(-t/2)
 1.0 t + 0.041666666666666664 t³ + 𝒪(t⁴)

julia> (exp(t/2) - exp(-t/2))/t
 1.0 + 0.041666666666666664 t² + 𝒪(t³)

# second order expansion
julia> t = Taylor1(Float64, 2)
 1.0 t + 𝒪(t³)

julia> exp(t/2) - exp(-t/2)
 1.0 t + 𝒪(t³)

julia> (exp(t/2) - exp(-t/2))/t
 1.0 + 𝒪(t²)
```

**Comment by matthieubulte** [comment]
Thanks, I will then wait for #248 to be merged then :)

**Comment by lbenet** [comment]
Closed by #248
